### DISCUSSION TITLE
Convert HeightmapUtils as Extensions

### DISCUSSION THREAD

**Discussion by IXLLEGACYIXL** [stride3d/stride#2573] [Ideas]
it would have benefits to convert the utils to extensions

I dont know why its separate, these are just helper methods anyway so it should be easy to convert them to extension methods
we can leave them in heightmaputils so no existing code breaks, or obsolete them and just put them into HeightMapExtensions or something like that

**Comment by Doprez** [comment]
Are you talking about the Physics heightmaps? I think those are tied to the physics colliders if I'm not mistake.

**Reply by IXLLEGACYIXL** [reply]
yes Stride.Physics.HeightMaps

and HeightMapUtils
like why is heightmaputils not an extension method and instead a separate class with heightmap parameter

**Reply by IXLLEGACYIXL** [reply]
i dont want to change logic etc, just make them extension methods if they will be still used

**Reply by Doprez** [reply]
Ah fair enough, I would definitely have a use myself Im assuming for the same reason as you with terrains?

I was previously using the ones from [TR.Stride](https://github.com/johang88/TR.Stride/blob/master/TR.Stride.Terrain/HeightmapExtensions.cs) but Im all for built in helpers if there are some to add.

**Reply by IXLLEGACYIXL** [reply]
i had some different use for heightmaps :D
they are pretty inaccurate for terrain tbh , the settlers 5 gave modders a card editor and i try to rebuild its functionalities into stride as it was pretty simple and had everything needed, and heightmaps dont fit that
